### PR TITLE
[Fix #5490] Add new `Lint/OrderedMagicComments` cop

### LINKED ISSUE #5490: Feature Request: Warn invalid location of coding pragma
coding pragma should be first line, or next of shebang.

```
%  cat /tmp/bad.rb
# frozen_string_literal: true
# coding: ascii
p [''.frozen?, ''.encoding]
%  ruby /tmp/bad.rb
[true, #<Encoding:UTF-8>]
%  cat /tmp/good.rb
# coding: ascii
# frozen_string_literal: true
p [''.frozen?, ''.encoding]
%  ruby /tmp/good.rb
[true, #<Encoding:US-ASCII>]
```

example: https://github.com/plataformatec/devise/pull/4753/files

### PR DESCRIPTION
**Pull request by koic**
Fixes #5490.

## Feature

This cop checks coding magic comment should be first line, or next of shebang.

### Bad case

```ruby
# frozen_string_literal: true
# coding: ascii
```

### Good case

```ruby
# coding: ascii
# frozen_string_literal: true
```

This cop checks only the order of magic comments.  This cop will not do what other cops have confirmed. (e.g. Such as checking whether Ruby version supports frozen string literal, etc.)

Also, this cop doesn't check whether shebang is at beginning of file. if shebang is not at beginning of file, executable script will not work as intended. IMO, to check if shebang is at beginning of file,
it may be better to make a cop for that.

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `rake default` or `rake parallel`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: config/enabled.yml [hunk 1]
```diff
@@ -590,6 +590,10 @@ Lint/NonLocalExitFromIterator:
   Description: 'Do not use return in iterator to cause non-local exit.'
   Enabled: true
 
+Lint/OrderedMagicComments:
+  Description: 'Checks coding magic comment should be first line, or next of shebang.'
```
Comment (bbatsov) [inline config/enabled.yml]: I'd suggest rewording this a bit. How about "Checks the proper ordering of magic comments and whether a magic comment is not placed before a shebang." or something along those lines.

File: spec/rubocop/cop/lint/ordered_magic_comments_spec.rb [hunk 2]
```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Lint::OrderedMagicComments, :config do
+  subject(:cop) { described_class.new(config) }
+
+  it 'registers an offense when using `frozen_string_literal` magic comment ' \
+     'is first line' do
+    expect_offense(<<-RUBY.strip_indent)
+      # frozen_string_literal: true
+      # coding: ascii
```
Comment (bbatsov) [inline spec/rubocop/cop/lint/ordered_magic_comments_spec.rb]: I guess you should also add some test for `encoding`, which is also valid.

File: lib/rubocop/cop/lint/ordered_magic_comments.rb [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    module Lint
+      #
+      # This cop checks coding magic comment should be first line,
+      # or next of shebang.
+      #
+      # @example
+      #   # bad
+      #
+      #   # frozen_string_literal: true
+      #   # coding: ascii
```
Comment (bbatsov) [inline lib/rubocop/cop/lint/ordered_magic_comments.rb]: This reminded me that I also using `encoding`, not `coding`. Probably we should add some check for this, although it's not a big deal in the age of `utf-8` by default.

File: lib/rubocop/cop/lint/ordered_magic_comments.rb [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      #
+      #   #!/usr/bin/env ruby
+      #   # coding: ascii
+      #   # frozen_string_literal: true
+      #   p [''.frozen?, ''.encoding] #=> [true, #<Encoding:US-ASCII>]
+      #
+      class OrderedMagicComments < Cop
+        include FrozenStringLiteral
+
+        MSG = 'Use coding magic comment should be first line, ' \
```
Comment (bbatsov) [inline lib/rubocop/cop/lint/ordered_magic_comments.rb]: I guess the message should just "The `coding` magic comment should precede all other magic comments." or something like this.

File: spec/rubocop/cop/lint/ordered_magic_comments_spec.rb (hunk 2 above)
Comment (koic) [reply spec/rubocop/cop/lint/ordered_magic_comments_spec.rb]: Thank you for review. I made the following changes by 99180fe.

- Improve description and offense message about this cop.
- Change `coding: ascii` to `encoding: ascii` and added some tests of `coding: ascii` and `-*- encoding : ascii-8bit -*-`.
